api/base_controller.go

```diff
@@ -273,3 +302,8 @@ func stripCredentials(ctx context.Context, object types.Object) {
 		log.C(ctx).Debugf("Object of type %s with id %s is not secured, so no credentials are cleaned up on response", object.GetType(), object.GetID())
 	}
 }
+
+func generateTokenForItem(obj types.Object) string {
+	nextPageToken := obj.(types.Pageable).GetPagingSequence()
```

Comment:
It looks like you expect all Objects to be Pageable. You could embed Pageable in Object and put the PagingSequence field as well as Pageable implementation in Base to avoid type casting here and code duplication accross all types

Reply:
The Notification type is not Pageable.

Reply:
Even if it implemented Pageable it wouldnt matter as GET notifications does not return a direct response (its a WS connection open call). If we had a normal GET API for notifications they would be pageble also so i think it would be fine if they implemented it